php-frontend/src/main/java/org/sonar/plugins/php/api/visitors/CheckContext.java

```diff
@@ -70,6 +70,16 @@ public interface CheckContext {
    */
   PreciseIssue newIssue(PHPCheck check, Tree startTree, Tree endTree, String message);
 
+  /**
+   * Report an issue with precise issue location based on lines and columns without relaying on tree elements
+   *
+   * @param check the instance of the rule for which issue should be created
+   * @param issueLocation issue location within a file including the location message
+   * @return issue with precise location
+   * @since 3.32
+   */
+  PreciseIssue newIssue(PHPCheck check, IssueLocation issueLocation);
```

Comment:
I'm not sure how strict we are in following semantic versioning, but if we are strict we should change the version from `3.32.0-SNAPSHOT` to `4.0.0-SNAPSHOT` too.

Reply:
For me, 4.0 will introduce major changes including breaking changes. This will only extend an API endpoint of adding issues.